### ISSUE TITLE
[BUG] LangChain design conflict: StructuredTool._run() missing 1 required keyword-only argument: 'config'

### ISSUE THREAD

**Issue by brycecf** [crewAIInc/crewAI-tools#86]
**Description**
Continuation of [#924](https://github.com/crewAIInc/crewAI/issues/924), which was closed without providing a long-term solution.

With `langchain-core==0.2.14`, a new required arg was added to StructuredTool._run() as shown [here](https://github.com/langchain-ai/langchain/blame/a72fddbf8d0c440d1ebc4bb4d5f924ce5205fc38/libs/core/langchain_core/tools.py#L919).  This breaks existing implementations using `crewai_tools.BaseTool`.

**Steps to Reproduce**
Create a custom tool using BaseTool.

**Expected behavior**
Elaborated on in [#924](https://github.com/crewAIInc/crewAI/issues/924), but this works with previous versions of `langchain-core`.

**Screenshots/Code snippets**
If applicable, add screenshots or code snippets to help explain your problem.

**Environment Details:**
- **Operating System**: [e.g., Ubuntu 20.04, macOS Catalina, Windows 10]
- **Python Version**: [e.g., 3.8, 3.9, 3.10]
- **crewAI Version**: [e.g., 0.30.11]
- **crewAI Tools Version**: [e.g., 0.2.6]

**Logs**
```
Thought: <LLM output>
Action: custom_crew
Action Input: {"query": "<LLM-derived query>"}


I encountered an error while trying to use the tool. This was the error: StructuredTool._run() missing 1 required keyword-only argument: 'config'.
```

**Possible Solution**
None

**Comment by joaomdmoura** [comment]
cc @gvieira

**Comment by joaomdmoura** [comment]
I think we fixed this on crewAI side right @gvieira , can you give us more context on what versions you are using just to be sure :)

**Comment by gvieira** [comment]
That's correct. We did fix it a while a go. A version update will make it right.
